app/services/find_a_job_service.rb

```
@@ -0,0 +1,75 @@
+class FindAJobService
+  attr_reader :api_id, :api_key, :query, :postcode, :distance, :path
+
+  BASE_URL = 'https://findajob.dwp.gov.uk/api/'.freeze
+  ResponseError = Class.new(StandardError)
+
+  def initialize(
+    api_id: Rails.configuration.find_a_job_api_id,
+    api_key: Rails.configuration.find_a_job_api_key,
+    options:
+  )
+    @api_id = api_id
+    @api_key = api_key
+    @query = options[:query]
+    @postcode = options[:postcode]
+    @distance = options[:distance]
+    @path = options[:path] || 'search'
+  end
+
+  def job_vacancies
+    JSON
+      .parse(response_body)
+      .dig('pager', 'total_entries')
+  end
+
+  def health_check
+    JSON.parse(perform_request)
+  end
+
+  private
+
+  def response_body
+    return '{}' unless api_id && api_key
```

I'm not sure about this behaviour. This is a required service so if no API key is defined we should see an error raised - I think this should raise an exception as per the pattern on the other services. If the service was optional then this implementation would be correct though.

---

Discussed offline and agreed to leave this as is.